src/main/java/xyz/nifeather/morph/network/multiInstance/master/MasterInstance.java

```diff
@@ -293,23 +293,25 @@ public void onSlaveRequestMetaSync(MIC2SRequestSyncCommand command)
             return;
         }
 
-        var cmd = new MIS2CSyncMetaCommand();
-
         var requestedUUIDs = command.requestedUUIDs;
 
         if (FeatherMorphMain.getInstance().debugOutputEnabled())
             logger.info("Client requesting %s UUIDs".formatted(requestedUUIDs.size()));
 
         if (requestedUUIDs.isEmpty()) return;
 
-        var disguises = morphManager.getRange(requestedUUIDs);
-
-        for (var meta : disguises)
-            cmd.appendMeta(new SocketPlayerMeta(Operation.ADD_IF_ABSENT, meta.getUnlockedDisguiseIdentifiers(), meta.uniqueId));
-
-        this.sendCommand(socket, cmd);
+        var dataStore = morphManager.getDataStore();
+        requestedUUIDs.forEach(uuid ->
+        {
+            dataStore.getOrLoad(uuid).thenAccept(meta ->
+            {
+                var cmd = new MIS2CSyncMetaCommand();
+                cmd.appendMeta(new SocketPlayerMeta(Operation.ADD_IF_ABSENT, meta.getUnlockedDisguiseIdentifiers(), uuid));
 
-        logMasterInfo("Synced %s metadata(s) to socket '%s'".formatted(disguises.size(), socket.getRemoteSocketAddress()));
+                this.sendCommand(socket, cmd);
+                logMasterInfo("Synced metadata for %s to socket '%s'".formatted(uuid, socket.getRemoteSocketAddress()));
+            });
+        });
```

Comment:
Maybe we should add some kind of batching here, so that we don't send them one by one